clang/lib/Sema/TreeTransform.h

```diff
@@ -10177,6 +10190,39 @@ TreeTransform<Derived>::TransformConstantExpr(ConstantExpr *E) {
   return TransformExpr(E->getSubExpr());
 }
 
+template <typename Derived>
+ExprResult
+TreeTransform<Derived>::TransformUniqueStableNameExpr(UniqueStableNameExpr *E) {
+  if (!E->isTypeDependent())
+    return E;
+  // We only have to handle the 'Expr' case, since the type case cannot be
+  // dependent.
```

Comment:
Should we assert this?

Reply:
I think the comment is wrong... types can be dependent as well, and we actually transform it below!  At one point I was writing this I forgot about "T::type" would be a TypeSourceInfo, and not an expression.